dtls.c

```diff
@@ -562,41 +563,51 @@ static const struct cipher_suite_param_t cipher_suite_params[] = {
   { TLS_ECDHE_ECDSA_WITH_AES_128_CCM_8,  8, DTLS_KEY_EXCHANGE_ECDHE_ECDSA },
   { TLS_ECDHE_ECDSA_WITH_AES_128_CCM,   16, DTLS_KEY_EXCHANGE_ECDHE_ECDSA },
 #endif /* DTLS_ECC */
-  /* TLS_NULL_WITH_NULL_NULL must always be the last entry as it
-   * indicates the stop marker for the traversal of this table. */
-  { TLS_NULL_WITH_NULL_NULL,             0, DTLS_KEY_EXCHANGE_NONE }
  };
 
-static inline const cipher_suite_param_t* get_cipher_suite_param(dtls_cipher_t cipher)
+static const int last_cipher_suite_param = sizeof(cipher_suite_params) / sizeof(cipher_suite_param_t);
+
+static inline int get_cipher_index(dtls_cipher_t cipher)
 {
-  const cipher_suite_param_t *param = cipher_suite_params;
-  while ((param->cipher_suite != cipher) &&
-         (param->cipher_suite != TLS_NULL_WITH_NULL_NULL)) {
-    param++;
+  for (int index = 0; index < last_cipher_suite_param ; ++index) {
+    if (cipher_suite_params[index].cipher_suite == cipher) {
+      return index;
+    }
   }
-  return param;
+  return 0;
 }
 
 /**
- * Get MAC length of cipher suite.
- * \param cipher cipher suite
- * \return MAC length of cipher. \c 0, if cipher is not supported.
+ * Get cipher suite.
+ * \param cipher_index index to cipher suite params
+ * \return cipher suite.
  */
-static inline int get_cipher_suite_mac_len(dtls_cipher_t cipher)
+static inline dtls_cipher_t get_cipher_suite(int cipher_index)
 {
-  return get_cipher_suite_param(cipher)->mac_length;
+  assert(0 <= cipher_index && cipher_index < last_cipher_suite_param);
```

Comment:
`0 <= cipher_index` can be omitted when the data types for the index have been changed to unsigned types.